src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -87,6 +88,234 @@ protected override Conversion GetInterpolatedStringConversion(BoundUnconvertedIn
+            switch (overloadResolutionResult.ValidResult.Member.ReturnType.SpecialType)
+            {
+                case SpecialType.System_Void:
+                    usesBoolReturn = false;
+                    break;
+                case SpecialType.System_Boolean:
+                    usesBoolReturn = true;
+                    break;
+                default:
+                    free();
+                    builderArguments = default;
+                    return false;
+            }
+
+            // We at least have an invocable, applicable TryFormatBaseString method on the type. We can proceed with finding the correct overloads for
+            // each component.
+
+            if (source.Parts.Length == 0)
+            {
+                free();
+                diagnostics.AddDiagnostics(source.Syntax, useSiteInfo);
+                builderArguments = ImmutableArray<MethodArgumentInfo>.Empty;
+                return true;
+            }
+
+            ArrayBuilder<MethodSymbol>? interpolationHoleCandidates = null;
+            var builderFormatCalls = ArrayBuilder<MethodArgumentInfo>.GetInstance(source.Parts.Length);
+
+            foreach (var part in source.Parts)
+            {
+                Debug.Assert(typeArguments.IsEmpty());
+                Debug.Assert(part is BoundLiteral or BoundStringInsert);
+                analyzedArguments.Clear();
+                overloadResolutionResult.Clear();
+
+                ArrayBuilder<MethodSymbol> candidateMethods;
+                if (part is BoundStringInsert insert)
+                {
+                    if (interpolationHoleCandidates is null && !tryGetCandidateMethods("TryFormatInterpolationHole", ref useSiteInfo, out interpolationHoleCandidates))
+                    {
+                        // PROTOTYPE(interp-string): We'll likely want to continue attempting to bind for errors when we're not directly being converted to a string
+                        free();
+                        builderFormatCalls.Free();
+                        builderArguments = default;
+                        return false;
+                    }
+
+                    candidateMethods = interpolationHoleCandidates;
+                    analyzedArguments.Arguments.Add(insert.Value);
+                    analyzedArguments.Names.Add(null);
+
+                    if (insert.Alignment is not null)
+                    {
+                        analyzedArguments.Arguments.Add(insert.Alignment);
+                        analyzedArguments.Names.Add(SyntaxFactory.IdentifierName("alignment"));
+                    }
+                    if (insert.Format is not null)
+                    {
+                        analyzedArguments.Arguments.Add(insert.Format);
+                        analyzedArguments.Names.Add(SyntaxFactory.IdentifierName("format"));
+                    }
+                }
+                else
+                {
+                    candidateMethods = baseStringCandidates;
```

Comment:
>candidateMethods [](start = 20, length = 16)

Do we really need to go through the trouble of doing overload resolution again for this case?